### ISSUE TITLE
Support S3 bucket access from users AWS hosts via vfs.s3.no_sign_request

### ISSUE THREAD

**Issue by atolopko-czi** [chanzuckerberg/cellxgene-census#278]
Currently, if a user is utilizing an AWS instance, the instance's IAM role is overriding the otherwise public access to the `s3://cellxgene-data-public` bucket, resulting in errors like:
```
tiledb.cc.TileDBError: [TileDB::S3] Error: Error while listing with prefix 's3://cellxgene-data-public/cell-census/2023-03-16/soma/__schema/' and delimiter '/'
Exception:  AccessDenied
Error message:  Access Denied
```

@ebezzi has theorized:
> bucket is public but if you have AWS credentials, then it will try to use them to access the bucket and override the "publicness"

The [fix](https://github.com/TileDB-Inc/TileDB/pull/3981) involves configuring the `vfs.s3.no_sign_request` TileDB option, which arriving in 2.15.1.  This will need to be pulled into a new TileDB-SOMA release and then can be addressed in Cell Census API.

Reminder:  make sure this is included in both the Python and R cellxgene census packages.

Original [Discussion](https://czi-sci.slack.com/archives/C013K0JB8F8/p1679339142559059).

**Comment by atolopko-czi** [comment]
The [s3 getobject docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html) state "If you grant READ access to the anonymous user, you can return the object without using an authorization header." Noting here, in case this allows us to solve the access issues w/o waiting for the `vfs.s3.no_sign_request` TileDB option.

**Comment by bkmartinjr** [comment]
Update: the core DB release [2.15.1](https://github.com/TileDB-Inc/TileDB/releases/tag/2.15.1) is out with the feature.  Now awaiting a tiledbsoma release incorporating this core DB version.

**Comment by pablo-gar** [comment]
@ebezzi can we cut a new release this coming week?

**Comment by ebezzi** [comment]
Still blocked, as the current `vfs.s3.no_sign_request` in TileDB 2.15.1 still requires credentials to be set up.

**Comment by ebezzi** [comment]
This has been resolved and release for Python. R work is still in progress.

**Comment by brianraymor** [comment]
@ebezzi to verify status and close if appropriate.

**Comment by ebezzi** [comment]
Fixed by #471.
